sockeye/average.py

```diff
@@ -155,6 +135,40 @@ def _read_metrics_points(path: str, metric: str) -> List[Tuple[float, int]]:
     return points
 
 
+def _strategy_best(points, size, maximize):
+    top_n = sorted(points, reverse=maximize)[:size]
```

Comment:
points are tuples of (metric_value, checkpoint) and here we are using the checkpoint ids as a tie-breaker, but the tie-breaker also depends on the maximize flag.
I think it should be implemented such that in case of metric_value ties, we tie-break by maximizing checkpoint id (i.e. choose the later one during training).